src/main/java/quacky/Deadline.java

```diff
@@ -0,0 +1,35 @@
+package quacky;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a deadline task in the Quacky application. A deadline task is a task that needs
+ * to be completed before a specific date.
+ */
+public class Deadline extends Task{
+    protected LocalDate by;
```

Comment:
Perhaps `byDate` would sound more like a noun for this variable? To obey code quality guidelines.